sound/soc/soc-topology.c

```diff
@@ -542,6 +542,19 @@ static void remove_link(struct snd_soc_component *comp,
 	kfree(link);
 }
 
+/* unload dai link */
+static void unload_link(struct snd_soc_component *comp,
+	struct snd_soc_dobj *dobj, int pass)
+{
+	if (pass != SOC_TPLG_PASS_LINK)
+		return;
```

Comment:
unclear to me why this is different from remove_link() ?

Reply:
@plbossart The remove_link() will handle the link which is created by soc_tplg_pcm_elems_load() and its type is SOC_TPLG_PASS_PCM_DAI actually. For SOC_TPLG_PASS_LINK type, we only config it but not create it. So, only unload is needed.